inference-engine/src/mkldnn_plugin/ngraph_transformations/reshape_1d_ops.cpp

```diff
@@ -97,23 +99,56 @@ std::shared_ptr<ngraph::Node> convert(const ngraph::Output<ngraph::Node> & data,
                                              node->get_auto_pad());
 }
 
+std::shared_ptr<ngraph::Node> add_reshape_before(const ngraph::Output<ngraph::Node>& data) {
+    auto shape_of = ngraph::op::util::make_try_fold<ngraph::opset1::ShapeOf>(data);
+
+    std::vector<std::int32_t> gather_values_first{ 0, 1 };
+    auto gather_indices_first = ngraph::opset1::Constant::create(
+        ngraph::element::i64,
+        ngraph::Shape{ gather_values_first.size() },
+        gather_values_first);
+
+    auto gather_axis = ngraph::opset1::Constant::create(ngraph::element::i64, ngraph::Shape{}, { 0 });
+    auto gather_first = ngraph::op::util::make_try_fold<ngraph::opset1::Gather>(shape_of, gather_indices_first, gather_axis);
+
+    std::vector<std::int32_t> gather_values_last{ 2 };
+    auto gather_indices_last = ngraph::opset1::Constant::create(
+        ngraph::element::i64,
+        ngraph::Shape{ gather_values_last.size() },
+        gather_values_last);
+    auto gather_last = ngraph::op::util::make_try_fold<ngraph::opset1::Gather>(shape_of, gather_indices_last, gather_axis);
+
+    auto unsqueezed_dimension = ngraph::opset1::Constant::create(ngraph::element::i64, ngraph::Shape{ 1 }, { 1 });
+
+    ngraph::NodeVector concatInputs = { gather_first, unsqueezed_dimension, gather_last };
+    auto concat = ngraph::op::util::make_try_fold<ngraph::opset1::Concat>(concatInputs, 0);
+    auto reshape = std::make_shared<ngraph::opset1::Reshape>(data, concat, true);
+    return reshape;
+}
+
+std::shared_ptr<ngraph::Node> add_reshape_after(const ngraph::Output<ngraph::Node>& data) {
+    auto shape_of = ngraph::op::util::make_try_fold<ngraph::opset1::ShapeOf>(data);
+
+    std::vector<std::int32_t> gather_values{ 0, 1, 3 };
+    auto gather_const = ngraph::opset1::Constant::create(ngraph::element::i64, ngraph::Shape{ gather_values.size() }, gather_values);
+    auto gather_axis = ngraph::opset1::Constant::create(ngraph::element::i64, ngraph::Shape{}, { 0 });
+    auto gather = ngraph::op::util::make_try_fold<ngraph::opset1::Gather>(shape_of, gather_const, gather_axis);
+    auto reshape = std::make_shared<ngraph::opset1::Reshape>(data, gather, true);
+    return reshape;
```

Comment:
you can replace it with Reshape(data, Const([0, 0, -1]), true)

Reply:
Done, thank you!